ansible_ai_connect/ai/api/wca/model_id_views.py

```diff
@@ -220,10 +221,12 @@ def do_validated_operation(request, api_key_provider, model_id_provider, on_succ
     event = None
     start_time = time.time()
     model_id = UNKNOWN_MODEL_ID
+
+    # An OrgId must be present
+    # See https://issues.redhat.com/browse/AAP-16009
+    organization = request._request.user.organization
+
     try:
-        # An OrgId must be present
-        # See https://issues.redhat.com/browse/AAP-16009
-        organization = request._request.user.organization
         if not organization:
```

Comment:
Retrieval of `organization` is kept within the `try` block to ensure the Segment Event is [sent](https://github.com/ansible/ansible-ai-connect-service/pull/1223/files#diff-f2f7efb68b77c2674b8507874c3eb91a12d3d992e643daeaa9e5d4ba70b8610fR277-R288). 

See tests:
- `test_model_id_views.test_validate_error_no_org_id`
- `test_model_id_views.test_set_key_without_org_id`

Reply:
not sure if I understand this: why are you moving this org-id retrieval outside of the `try`?

Reply:
Moved back. The new `expect` block needed `organization` however I took a different approach.